source/io/HttpClient.cpp

```diff
@@ -88,16 +135,17 @@ VIREO_FUNCTION_SIGNATURE6(HttpClientOpen, StringRef, StringRef, StringRef, UInt3
         return THREAD_EXEC()->Stop();
     }
 
+    TypeRef typeRefUInt32 = TypeManagerScope::Current()->FindType("UInt32");
+    TypeRef typeRefErrorCluster = TypeManagerScope::Current()->FindType("ErrorCluster");
```

Comment:
I was looking at "TypeManager::New" and "TypeDefiner::DefineStandardTypes", perhaps we can come up with a caching mechanism to access the most commonly used type definitions. But I think that would be part of another pull request and not this one.

Reply:
Also we can use StaticTypeAndData type on the Vireo function definition. That will give you access to both the type and a pointer to the data. You might want to do one function to see how it will look. The problem with that is that the Vireo definition for the instruction will allow more types than intended. Again this is more likely a refactor to do in another pull request.

Reply:
I agree we can continue refactoring but I think this is a good chunk of work for this PR.